frameworks/non-keyed/kobold/src/lib.rs

```diff
@@ -0,0 +1,226 @@
+            label.push(' ');
+            label.push_str(colour);
+            label.push(' ');
+            label.push_str(noun);
+            self.rows.push(RowData {
+                id: i + d,
+                label,
+            });
+        }
+        self.last += count;
+    }
+
+    fn remove(&mut self, id: usize) {
+        self.rows.remove(id);
+    }
+
+    fn run(&mut self) {
+        self.clear();
+        self.append_rows(1_000)
+    }
+    fn run_lots(&mut self) {
+        self.clear();
+        self.append_rows(10_000)
+    }
+    fn add(&mut self) {
+        self.append_rows(1_000)
+    }
+    fn update(&mut self) {
+        self.rows.iter_mut().step_by(10).for_each(|row| {
+            row.label.reserve_exact(4);
+            row.label.push_str(" !!!")
+        });
+    }
+    fn clear(&mut self) {
+        self.rows = Vec::new();
+        self.selected = None
+    }
+    fn swap(&mut self) {
+        if self.rows.len() > 998 {
+            self.rows.swap(1, 998);
+        }
+    }
+    fn select(&mut self, id: usize) {
+        self.selected = Some(id)
+    }
+}
+
+#[component]
+fn Row(num: usize, row: RowData, state: &Hook<State>) -> impl View {
+    let is_in_danger = class!("danger" if state.selected == Some(row.id));
+
+    bind! { state:
+        let remove = move |_| state.remove(num);
+        let select = move |_| state.select(row.id);
+    }
+
+    view! {
+        <tr.{is_in_danger}>
+            <td class="col-md-1">{row.id.to_string()}</td>
+            <td class="col-md-4"><a onclick={select}>{row.label}</a></td>
+            <td class="col-md-1"><a onclick={remove}><span class="glyphicon glyphicon-remove" aria-hidden="true"></span></a></td>
+            <td class="col-md-6"/>
+        </tr>
+    }
+}
+
+enum ButtonAction {
+    Run,
+    RunLots,
+    Add,
+    Update,
+    Clear,
+    Swap,
+}
+
+#[component]
+fn Button(action: ButtonAction, state: &Hook<State>) -> impl View {
+    let (id, text) = match action {
+        ButtonAction::Run => ("run", "Create 1,000 rows"),
+        ButtonAction::RunLots => ("runlots", "Create 10,000 rows"),
+        ButtonAction::Add => ("add", "Append 1,000 rows"),
+        ButtonAction::Update => ("update", "Update every 10th row"),
+        ButtonAction::Clear => ("clear", "Clear"),
+        ButtonAction::Swap => ("swaprows", "Swap Rows"),
+    };
+
+    bind! { state:
+        let onclick = move |_| match action {
+            ButtonAction::Run => state.run(),
+            ButtonAction::RunLots => state.run_lots(),
+            ButtonAction::Add => state.add(),
+            ButtonAction::Update => state.update(),
+            ButtonAction::Clear => state.clear(),
+            ButtonAction::Swap => state.swap(),
+        };
+    }
+
+    view! {
+        <div class="col-sm-6 smallpad">
+        <button
+            id={id}
+            class="btn btn-primary btn-block"
+            type="button"
+            {onclick}
+        >
+            { text }
```

Comment:
```suggestion
            { static text }
```

This `text` never changes. Alternative would be to wrap the entire body of `Button` in `fence` on `ButtonAction`, though that probably would require a  `#[derive(Diff)]` to be user friendly (note to self).

Reply:
This compiles fine but rust-analyzer complains errors [no such field](https://rust-analyzer.github.io/manual.html#no-such-field). Do you have any idea why?